### ISSUE TITLE
t/01_modules.t is problematic

### ISSUE THREAD

**Issue by eserte** [mhasch/perl-Bundle-Maintainer-MHASCH#1]
The test suite usually fails on certain CPAN Testers setups e.g. on test-only setups, for example like this (list of missing modules usually varies):
```
#   Failed test 'require Math::Logic::Ternary;'
#   at t/01_modules.t line 46.
#     Tried to require 'Math::Logic::Ternary'.
#     Error:  Can't locate Math/Logic/Ternary.pm in @INC (you may need to install the Math::Logic::Ternary module) (@INC contains: /usr/home/cpansand/.cpan/build/2021022707/Bundle-Maintainer-MHASCH-0.004-6/blib/lib /usr/home/cpansand/.cpan/build/2021022707/Bundle-Maintainer-MHASCH-0.004-6/blib/arch /usr/perl5.28.1Dp/lib/site_perl/5.28.1/amd64-freebsd-ld /usr/perl5.28.1Dp/lib/site_perl/5.28.1 /usr/perl5.28.1Dp/lib/5.28.1/amd64-freebsd-ld /usr/perl5.28.1Dp/lib/5.28.1 .) at (eval 56) line 2.

#   Failed test 'require Math::Polynomial::Multivariate;'
#   at t/01_modules.t line 46.
#     Tried to require 'Math::Polynomial::Multivariate'.
#     Error:  Can't locate Math/Polynomial/Multivariate.pm in @INC (you may need to install the Math::Polynomial::Multivariate module) (@INC contains: /usr/home/cpansand/.cpan/build/2021022707/Bundle-Maintainer-MHASCH-0.004-6/blib/lib /usr/home/cpansand/.cpan/build/2021022707/Bundle-Maintainer-MHASCH-0.004-6/blib/arch /usr/perl5.28.1Dp/lib/site_perl/5.28.1/amd64-freebsd-ld /usr/perl5.28.1Dp/lib/site_perl/5.28.1 /usr/perl5.28.1Dp/lib/5.28.1/amd64-freebsd-ld /usr/perl5.28.1Dp/lib/5.28.1 .) at (eval 69) line 2.
# Looks like you failed 2 tests of 8.
t/01_modules.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/8 subtests 
```
This can be reproduced using the following commands with a "virgin" perl (without any prereqs installed):
```
cpan
test MHASCH/Bundle-Maintainer-MHASCH-0.004.tar.gz
```
Bundles are handled specially by CPAN.pm and really work only for installs, it seems. I think it's best to make the failing test script an author-only test, or convert the Bundle into a Task.

**Comment by mhasch** [comment]
Slaven,

Thanks for the suggestion. That one could want to run the test suite without installing the bundle contents first was new to me.
I saw the module loading test as a service for users to check whether the somewhat obscure bundle mechanics actually worked to deploy these modules. Thus, testing maintainer-only would be beside the point. I'd rather like to identify environments purposefully not installing bundled modules and skip the tests just there. For the time being, i'll make the tests non-fatal, so I can see test results but don't trigger more testers into investigating the matter.

-Martin

**Comment by mhasch** [comment]
Fixed in Bundle-Maintainer-MHASCH-0.005.
